qiita_db/metadata_template.py

```diff
@@ -1445,6 +1462,89 @@ def extend(self, md_template):
+        new_map = self._clean_validate_template(md_template, self.id,
+                                                conn_handler)
+        # Retrieving current metadata
+        current_map = self._transform_to_dict(conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.{0} WHERE {1}=%s".format(self._table,
+                                                          self._id_column),
+            (self.id,)))
+        dyn_vals = self._transform_to_dict(conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.{0}".format(self._table_name(self.id))))
+
+        for k in current_map:
+            current_map[k].update(dyn_vals[k])
+            try:
+                del current_map[k]['study_id']
+            except KeyError:
+                pass
+
+        # converting sql results to dataframe
+        current_map = pd.DataFrame.from_dict(current_map, orient='index')
+
+        # simple validations of sample ids and column names
+        samples_diff = set(
+            new_map.index.tolist()) - set(current_map.index.tolist())
+        if samples_diff:
+            raise QiitaDBError('The new sample template differs from what is '
+                               'stored in database by these samples names: %s'
+                               % ', '.join(samples_diff))
+        columns_diff = set(new_map.columns) - set(current_map.columns)
+        if columns_diff:
+            raise QiitaDBError('The new sample template differs from what is '
+                               'stored in database by these columns names: %s'
+                               % ', '.join(columns_diff))
+
+        # here we are comparing two dataframes following:
+        # http://stackoverflow.com/questions/24495695
+        current_map.sort(axis=0, inplace=True)
+        current_map.sort(axis=1, inplace=True)
+        new_map.sort(axis=0, inplace=True)
+        new_map.sort(axis=1, inplace=True)
+        map_diff = (current_map != new_map).stack()
+        map_diff = map_diff[map_diff]
+        map_diff.index.names = ['id', 'column']
+        changed_cols = map_diff.index.get_level_values('column').unique()
+
+        for col in changed_cols:
+            self.update_category(col, new_map[col].to_dict())
+
+        # figuring out the filepath of the backup
+        _id, fp = get_mountpoint('templates')[0]
+        fp = join(fp, '%d_%s.txt' % (self.id, strftime("%Y%m%d-%H%M%S")))
+        # storing the backup
+        self.to_file(fp)
+
+        # adding the fp to the object
+        self.add_filepath(fp)
+
+        # generating all new QIIME mapping files
+        for rd_id in Study(self.id).raw_data():
+            for pt_id in RawData(rd_id).prep_templates:
+                pt = PrepTemplate(pt_id)
+                for _, fp in pt.get_filepaths():
+                    # the difference between a prep and a qiime template is the
+                    # word qiime within the name of the file
+                    if 'qiime' not in fp:
```

Comment:
Can you check that `_qiime_` is in the `basename` of the `fp`? We could have a pathological case where someone's filesystem's name is `/qiime/` and then all paths would be picked-up by this if statement.